webhooks/virtualmachine/validation/virtualmachine_validator.go

```diff
@@ -1568,3 +1572,59 @@ var capvDefaultServiceAccount = regexp.MustCompile("^system:serviceaccount:svc-t
 func isCAPVServiceAccount(username string) bool {
 	return capvDefaultServiceAccount.Match([]byte(username))
 }
+
+func (v validator) validateChecks(
+	ctx *pkgctx.WebhookRequestContext,
+	newVM, oldVM *vmopv1.VirtualMachine) field.ErrorList {
+
+	var allErrs field.ErrorList
+
+	if errs := v.validateChecksPowerOn(ctx, newVM, oldVM); len(errs) > 0 {
+		allErrs = append(allErrs, errs...)
+	}
+
+	return allErrs
+}
+
+func (v validator) validateChecksPowerOn(
+	ctx *pkgctx.WebhookRequestContext,
+	newVM, oldVM *vmopv1.VirtualMachine) field.ErrorList {
+
+	const prefix = vmopv1.CheckAnnotationPowerOn + "/"
+	var allErrs field.ErrorList
+
+	if ctx.IsPrivilegedAccount {
+		return allErrs
+	}
+
+	for k := range oldVM.Annotations {
+		if strings.HasPrefix(k, prefix) {
+			f := field.NewPath("metadata", "annotations", k)
+			_, onOldVM := oldVM.Annotations[k]
+			_, onNewVM := newVM.Annotations[k]
+
+			if onOldVM && !onNewVM {
+				allErrs = append(allErrs, field.Forbidden(f, delRestrictedAnnotation))
+			}
+		}
+	}
+
+	for k := range newVM.Annotations {
+		if strings.HasPrefix(k, prefix) {
+			f := field.NewPath("metadata", "annotations", k)
+			oldVal, onOldVM := oldVM.Annotations[k]
+			newVal, onNewVM := newVM.Annotations[k]
+
+			switch {
+			case onOldVM && !onNewVM:
+				allErrs = append(allErrs, field.Forbidden(f, delRestrictedAnnotation))
```

Comment:
Same here. I don't think this would ever be the case because onNewVM is always true as you're looping through new VM in this loop?

Reply:
Good catch, fixed!